src/libraries/System.Private.CoreLib/src/System/MathF.cs

```diff
@@ -429,6 +429,35 @@ public static float Round(float x, int digits)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static float Round(float x, MidpointRounding mode)
         {
+            if (AdvSimd.IsSupported)
+            {
+                // Inline the most common cases
+                if (mode == MidpointRounding.ToEven)
+                {
+                    return Round(x);
+                }
+                if (mode == MidpointRounding.AwayFromZero)
+                {
+                    return AdvSimd.RoundAwayFromZero(Vector64.CreateScalarUnsafe(x)).ToScalar();
+                }
+            }
+            if (Sse41.IsSupported)
+            {
+                // Inline the most common cases
+                if (mode == MidpointRounding.ToEven)
+                {
+                    return Round(x);
+                }
+                if (mode == MidpointRounding.AwayFromZero)
+                {
+                    Vector128<float> xVec = Vector128.CreateScalarUnsafe(x);
+                    Vector128<float> tmp = Sse.And(xVec, Vector128.Create(0x8000000000000000UL).AsSingle());
+                    tmp = Sse.Or(tmp, Vector128.Create(0x3fdfffffffffffffUL).AsSingle());
```

Comment:
This is going to be wrong for `SSE` as its not the binary representation of `BitDecrement(0.5f)` for float

I'm also not sure that this implementation is correct in the face of all representable floats, I'm pretty sure there is going to be rounding and precision wonkiness in a number of corner cases.

Reply:
> This is going to be wrong for `SSE` as its not the binary representation of `BitDecrement(0.5f)` for float
> 

yeah, copy-paste